### ISSUE TITLE
I/O customization rule not run on split sub-object of a non-collection object.

### ISSUE THREAD

**Issue by pcanal** [root-project/root#8428]
This is seen in https://cdcvs.fnal.gov/redmine/issues/25893 and is triggered by a change in CLHEP::Hep3Vector (from 3 individual members to a single array) that is stored in a sub-object of a simple object (no collection involved, it works in that case).

**Comment by jblomer** [comment]
@pcanal Redmine is behind a login wall.  Could you paste the core of the bug report in this issue, too?

**Comment by pcanal** [comment]
Synopsis of the original error report:

Description

I had thought that our `art v3_09_01` based art passed all of our validation tests but I missed a single histogram that failed. I discovered it yesterday.

I traced it back to the following. We have a file

`/pnfs/mu2e/scratch/users/brownd/workflow/default/outstage/44170899/00/00000/dig.brownd.CeEndpointMix.MDC2020b.001201_00000000.art`

It was written by `art v3_06_03`. One of the data products is of type `std::vector<StepPointMC>`, where `StepPointMC` is a class in our code base. The class contains 4 data members of type `CLHEP::Hep3Vector`:

https://github.com/Mu2e/Offline/blob/da4772bcb99f7b42bd401c3b15d56a211124553a/MCDataProducts/inc/StepPointMC.hh#L209

The `art::InputTag` is `"compressDigiMCs:virtualdetector"` . When I write a small job using `art v3_09_01` to read this data product and print out it's data members, all 4 of the `Hep3Vector` are `(0,0,0)`.

When I run the same job using `art v3_06_03` all of the printout is correct.

Remember that between `art v3_06_03` and `art v3_09_01` there was a schema change in `CLHEP::Hep3Vector`. You already fixed one bug related to that.

I will prepare a stripped down example and add it to this issue.

To reproduce. Unwind the following tar file on a machine that sees `/pnfs/mu2e and /cvmfs/mu2e.opensciencegrid.org`
```
tar xzf /mu2e/app/users/kutschke/Bugs/Hep3Vector/artv3_09.tar.gz  
cd artv3_09
source /cvmfs/mu2e.opensciencegrid.org/setupmu2e-art.sh
source setup.sh
scons -j 8
mu2e -c bug.fcl
```
This prints most of the content of the the `StepPointMCs`. The bug is that the 4 `Hep3Vectors` are **all zero**. It's OK that the non-ionizing and visual energy depositions are zero; this data product captures steps in a virtual detector embedded inside a volume with a vacuum of `10^{-4} torr`.

This uses the deprecated Mu2e "Satellite release" system to build on top of the Offline tag v10_00_00 on /cvmfs/mu2e.opensciencegrid.org/Offline/v10_00_00/SLF7/prof/Offline .
